doc/en/components/grids/hierarchical-grid/load-on-demand.md

```diff
@@ -0,0 +1,925 @@
+    });
+
+    orderDetailsRowIsland.addEventListener("gridCreated", (event: any) => {
+        this.gridCreated(event, "OrderID");
+    });
+
+    hierarchicalGrid.isLoading = true;
+
+    getData({ parentID: null, rootLevel: true, key: "Customers" }).then((data: any) => {
+        hierarchicalGrid.isLoading = false;
+        hierarchicalGrid.data = data;
+        hierarchicalGrid.markForCheck();
+    });
+}
+
+public gridCreated(event: CustomEvent<IgcGridCreatedEventArgs>, _parentKey: string) {
+    const context = event.detail;
+    const dataState = {
+        key: context.owner.childDataKey,
+        parentID: context.parentID,
+        parentKey: _parentKey,
+        rootLevel: false
+    };
+
+    context.grid.isLoading = true;
+
+    getData(dataState).then((data: any[]) => {
+        context.grid.isLoading = false;
+        context.grid.data = data;
+        context.grid.markForCheck();
+    });
+}
+```
+<!-- end: WebComponents -->
+
+<!-- React -->
+```tsx
+const hierarchicalGrid = useRef<IgrHierarchicalGrid>(null);
+
+useEffect(() => {
+    hierarchicalGrid.current.isLoading = true;
+    
+    getData({ parentID: null, rootLevel: true, key: "Customers" }).then(
+      (data: any) => {
+        hierarchicalGrid.current.isLoading = false;
+        hierarchicalGrid.current.data = data;
+        hierarchicalGrid.current.markForCheck();
+    }
+  );
+}, []);
+
+function gridCreated(rowIsland: IgrRowIsland, event: IgrGridCreatedEventArgs, _parentKey: string) {
+    const context = event.detail;
+    const dataState = {
+        key: rowIsland.childDataKey,
+        parentID: context.parentID,
+        parentKey: _parentKey,
+        rootLevel: false,
+    };
+    
+    context.grid.isLoading = true;
+    
+    getData(dataState).then((data: any[]) => {
+        context.grid.isLoading = false;
+        context.grid.data = data;
+        context.grid.markForCheck();
+    });
+}
+```
+<!-- end: React -->
+
+<!-- Blazor -->
+```razor
+igRegisterScript("OnGridRendered", () => {
+    const grid = document.getElementById("hGrid");
+
+    grid.isLoading = true;
+
+    getData({ parentID: null, rootLevel: true, key: "Customers" }).then(
+        (data) => {
+            grid.isLoading = false;
+            grid.data = data;
+            grid.markForCheck();
+        });
+}, false)
+
+igRegisterScript("OnGridCreated", (args) => {
+    const context = args.detail;
+    const _parentKey = context.owner.childDataKey === "Orders" ? "CustomerID" : "OrderID";
+    const dataState = {
+        key: context.owner.childDataKey,
+        parentID: context.parentID,
+        parentKey: _parentKey,
+        rootLevel: false,
+    };
+
+    context.grid.isLoading = true;
+
+    getData(dataState).then((data) => {
+        context.grid.isLoading = false;
+        context.grid.data = data;
+        context.grid.markForCheck();
+    });
+}, false)
+```
+<!-- end: Blazor -->
+
+### API References
+
+* `HierarchicalGrid`
+* `RowIsland`
```

Comment:
We might want to update the mappings for these two references so they lead to the API Docs